logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt

```diff
@@ -213,13 +222,35 @@ class UserDataSource(
         userDAO.updateUserAvailabilityStatus(getSelfUserIDEntity(), availabilityStatusMapper.fromModelAvailabilityStatusToDao(status))
     }
 
+    override suspend fun updateOtherUserAvailabilityStatus(userId: UserId, status: UserAvailabilityStatus) {
+        userDAO.updateUserAvailabilityStatus(idMapper.toDaoModel(userId), availabilityStatusMapper.fromModelAvailabilityStatusToDao(status))
+    }
+
     override suspend fun getAllKnownUsersNotInConversation(conversationId: ConversationId): Either<StorageFailure, List<OtherUser>> {
         return wrapStorageRequest {
             userDAO.getUsersNotInConversation(idMapper.toDaoModel(conversationId))
                 .map { publicUserMapper.fromDaoModelToPublicUser(it) }
         }
     }
 
+    override suspend fun getUsersFromTeam(teamId: TeamId): Either<StorageFailure, List<OtherUser>> {
+        return wrapStorageRequest {
+            val selfUserId = getSelfUserIDEntity()
+
+            userDAO.getAllUsersByTeam(teamId.value)
+                .filter { it.id != selfUserId }
+                .map(publicUserMapper::fromDaoModelToPublicUser)
+        }
+    }
+
+    override suspend fun getTeamRecipients(teamId: TeamId): Either<CoreFailure, List<Recipient>> =
+        getUsersFromTeam(teamId)
+            .map { users -> users.map { idMapper.toApiModel(it.id) } }
+            .flatMap {
+                wrapApiRequest { clientApi.listClientsOfUsers(it) }
```

Comment:
Is this really necessary? Can't we just use what we have from the database? We have a `ClientDAO`?

Reply:
it requires some additional changes, but good point :)